src/ActionsMenu.js

```diff
@@ -0,0 +1,141 @@
+            },
+            left180: {
+                isAllowed: false,
+                labelKey: "Command.left180"
+            },
+            right45: {
+                isAllowed: true,
+                labelKey: "Command.right45"
+            },
+            right90: {
+                isAllowed: true,
+                labelKey: "Command.right90"
+            },
+            right180: {
+                isAllowed: false,
+                labelKey: "Command.right180"
+            }
+        }
+    }
+    constructor (props: ActionsMenuProps) {
+        super(props);
+        this.state = { showMenu: false };
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <div className='ActionsMenu__header'>
+                    <h2 className='ActionsMenu__header-heading'>
+                        <FormattedMessage id='ActionsMenu.title' />
+                    </h2>
+                    <ActionsMenuToggle
+                        className='ActionsMenu__header-toggle'
+                        intl={this.props.intl}
+                        editingDisabled={!!this.props.editingDisabled}
+                        onClick={this.showHideMenu}
+                    />
+
+                    { (!this.props.editingDisabled && this.state.showMenu) ? this.generateMenu(): undefined}
+                </div>
+
+            </React.Fragment>
+        );
+    }
+
+    /* istanbul ignore next */
+    handleKeyDown = (event: SyntheticKeyboardEvent<HTMLElement>) => {
+        if (event.key === 'Escape') {
+            this.showHideMenu();
+        }
+    }
+
+    showHideMenu = () => {
+        if (!this.props.editingDisabled) {
+            this.setState((state) => {
+                return { showMenu: !(state.showMenu)}
+            });
+        }
+    }
+
+    generateMenu = () => {
+        const actionsMenuItems = [];
+        // TODO: Discuss how to evolve this into a deeper structure when we add groups and things other than actions.
+        Object.keys(this.props.menuItems).forEach((itemKey) => {
+            const isAllowed: boolean = !!this.props.allowedActions[itemKey];
+            const isUsed: boolean = !!this.props.usedActions[itemKey];
+            // TODO: Add a mechanism for values to come back to us.
+            const itemChangeHandler = (event: Event) => {
+                if (this.props.changeHandler) {
+                    this.props.changeHandler(event, itemKey);
+                }
+            };
+            actionsMenuItems.push(
+                <ActionsMenuItem
+                    intl={this.props.intl}
+                    isAllowed={isAllowed}
+                    isUsed={isUsed}
+                    itemKey={itemKey}
+                    key={itemKey}
+                    onChange={itemChangeHandler}
+                />
+            );
+        });
+        return (<div className="ActionsMenu__menu" onBlur={this.showHideMenu} onKeyDown={this.handleKeyDown}>
```

Comment:
Using showHideMenu onBlur causes the menu to close when tabbing through the menu items, because tab key fires onBlur event. We should implement separate onBlur function to know not to close the menu on tab key.

Reply:
I ended up having to check to see if the relatedTarget for the blur was one of the menu's children, similar to what we had to do with the drag and drop on the `ProgramBlockEditor`.  This appears to work as expected, i.e. tabbing out of the menu closes it, but tabbing within does not.

Reply:
I eventually ended up completely getting rid of my one-off blur handler in favour of using a `FocusTrapManager`.